### PR TITLE
feat: add security policy for api gateway (secuirty compliance)

### PR DESCRIPTION
**Pull request by raiventeguh**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -144,3 +144,9 @@ variable "cloudflare_zone_id" {
   type        = string
   default     = null
 }
+
+variale "security_policy" {
```
Comment (doskoyz) [inline variables.tf:148]: typo

File: apigateway.tf [hunk 2]
```diff
@@ -62,9 +62,10 @@ resource "aws_api_gateway_domain_name" "domain" {
 }
 
 resource "aws_api_gateway_deployment" "stage" {
-  count       = local.use_api_gateway
-  rest_api_id = aws_api_gateway_rest_api.api-gateway[0].id
-  stage_name  = var.environment
+  count           = local.use_api_gateway
+  rest_api_id     = aws_api_gateway_rest_api.api-gateway[0].id
+  stage_name      = var.environment
+  security_policy = var.security_policy
```
Comment (doskoyz) [inline apigateway.tf:68]: are you sure this resource have this parameter?
I dont see it in https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_deployment

File: variables.tf (hunk 1 above)
Comment (raiventeguh) [reply variables.tf:148]: where is the type? 
it's security_policy

Comment (raiventeguh) [reply variables.tf:148]: oh variable

File: apigateway.tf [hunk 3]
```diff
@@ -62,9 +63,9 @@ resource "aws_api_gateway_domain_name" "domain" {
 }
 
 resource "aws_api_gateway_deployment" "stage" {
-  count       = local.use_api_gateway
-  rest_api_id = aws_api_gateway_rest_api.api-gateway[0].id
-  stage_name  = var.environment
+  count           = local.use_api_gateway
+  rest_api_id     = aws_api_gateway_rest_api.api-gateway[0].id
+  stage_name      = var.environment
```
Comment (doskoyz) [inline apigateway.tf:69]: extra space here

**Review by doskoyz — APPROVED** [review]

**Pull request merged**